http/src/config.rs

```diff
@@ -52,27 +84,24 @@ pub fn initialize(
 fn create(
     config: File,
     bits: NonZeroU16,
-    profiles: Vec<String>,
+    profiles: Vec<Profile>,
 ) -> Result<(), InitializationError> {
     use multibase::Base::Base64Pad;
     use prost::Message;
     use std::io::BufWriter;
 
+    let api_addrs = match profiles[0] {
+        Profile::Test => "127.0.0.1:0",
+        Profile::Default => "127.0.0.1:4004",
```

Comment:
I think string literals aren't needed here `([127, 0, 0, 1], 0)` could probably be used, or at least could be parsed following this `match` or on the match arms? Or a [`multiaddr!`](https://docs.rs/parity-multiaddr/0.9.3/parity_multiaddr/macro.multiaddr.html) macro could produce the required multiaddr, not sure which way it should be done to have it the easiest.

Reply:
I initially started out with a `MultiAddr` though switched to parsing to a `SocketAddr` as it was the simplest to use with warp; otherwise I needed to implement `Into<SocketAddr>` for `Protocol` if I remember correctly. Would this be preferred?

Reply:
Actually the go implementation uses `MultiAddr` as well as a capitalised `API` key:
`"API": "/ip4/127.0.0.1/tcp/5001"